streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -713,6 +723,44 @@ boolean runLoop() {
         return true;
     }
 
+    // visible for testing
+    void maybeGetClientInstanceIds() {
+        // we pass in a timeout of zero into each `clientInstanceId()` call
+        // to just trigger the "get instance id" background RPC;
+        // we don't want to block the stream thread that can do useful work in the meantime
+
+        if (fetchDeadlineClientInstanceId != -1) {
+            if (!mainConsumerInstanceIdFuture.isDone()) {
+                if (fetchDeadlineClientInstanceId >= time.milliseconds()) {
+                    try {
+                        mainConsumerInstanceIdFuture.complete(mainConsumer.clientInstanceId(Duration.ZERO));
+                        maybeResetFetchDeadline();
+                    } catch (final IllegalStateException disabledError) {
+                        mainConsumerInstanceIdFuture.complete(null);
```

Comment:
why are we ignored ISE?

Reply:
ISE is throws if telemetry is disabled on the client -- in general, we assume that it's enabled for all clients, or disabled for all clients, but it could also be, that it's disabled for _some_ client only. For the last case, we want to swallow the error what happens here (on general, we say, a single error on any client fails the call to `KafkaStreams#clientInstanceIds()` but for this case we need to make an exception).

Let me know if you agree or disagree. Will add a comment. (Seems `disabledError` as var name is not descriptive enough).

Reply:
Sounds good. Yeah a comment will help here